filecoin-virtual-machine/section-a.md

```diff
@@ -0,0 +1,307 @@
+
+More importantly, this enables developers to build novel applications. Imagine a dapp or DAO that can be built to incentivize, analyze and store upload metadata on-chain. There are a couple of examples of this:
+
+- Rewarding $TOKEN based on the upload of a particular file and their CID.
+- Being able to track CIDs and deal IDs onchain for verification and airdropping.
+- Building more advanced, robust DataDAOs (check out the starter kit [here](https://github.com/filecoin-project/fevm-data-dao-kit)!)
+
+## Appendix
+
+For more information, check out the following code examples to upload files:
+
+1. [NodeJS Code Examples](../lighthouse-sdk/code-examples/nodejs-backend/)
+2. [Frontend(React, Next..) Code Examples](../lighthouse-sdk/code-examples/browser-frontend/)
+3. [Lighthouse File](https://files.lighthouse.storage/)
+
+You can also check out the following documentation to learn more about various other jobs you can register
+
+**Note**: Deal by default using SDK will go to the mainnet unless deal parameters are provided mentioned [here](../javascript/functions/upload.md).
+
+### Appendix: Flow diagram
+
+A full flow diagram of the lighthouse SDK can be found below:
+
+<figure><img src="../.gitbook/assets/Screenshot 2023-07-20 153056.png" alt=""><figcaption></figcaption></figure>
+
+### Appendix: Deal Parameters
+
+When uploading a file, you can customize how it's stored in Lighthouse using the **deal parameters**:
+
+**num\_copies**: Decide how many backup copies you want for your file. The Max limit is 3. For instance, if set to 3, your file will be stored by 3 different storage providers.
+
+**repair\_threshold**: Determines the time a storage sector is considered "broken" if a provider fails to confirm they still have your file. It's measured in "epochs", with 28800 epochs being roughly 10 days.
+
+**renew\_threshold**: Specifies when your storage deal should be renewed. It's also measured in epochs.
+
+**miner**: If you have preferred miners, list their addresses here. For testing, it's recommended to use t017840.
+
+**network**: This should always be set to 'calibration' (for RAAS services to function) unless you want to use the mainnet.
+
+**add\_mock\_data**: This field is used to make smaller files reach the minimum file size accepted on the Lighthouse calibration test network (1 MB). If your file is less than the minimum size, `add_mock_data` will append a mock file to ensure it meets the storage requirements. The value indicates the size in MB. For instance, if your file is 256KB, the add\_mock\_data should be set to 2 to the minimum target.
+
+Example:
+
+```
+// Sample JSON of deal parameters
+const dealParams = {
+  num_copies: 2,
+  repair_threshold: 28800,
+  renew_threshold: 240,
+  miner: ["t017840"],
+  network: 'calibration',
+  add_mock_data: 2
+};
+```
+
+```
+# Use cases
```

Comment:
This doesn't describe what dealParam_default, dealParam_mock, etc are. Use the code block here directly instead: https://docs.google.com/document/d/1vFeIUx9sjw-DMB9goZwNAvrnzyiJJ6XYUMap5W15L8w/edit#bookmark=id.xl2pdmf1jndr